### PR TITLE
makefile helpers

### PR DESCRIPTION
**Pull request by Alexsandruss**
# Description

Changes proposed in this pull request:
- Autodetection of platform using `uname`
- Removal of platform arg. from CI and other scripts 
- Generation of cmake configs in makefile if cmake is available

### CODE REVIEW

**Review by KulikovNikita — APPROVED** [review]

File: makefile [hunk 1]
```diff
@@ -18,6 +18,10 @@
 # Common macros
 #===============================================================================
 
+ifeq ($(PLAT),)
+    PLAT:=$(shell bash dev/make/identify_os.sh)
```
Comment (napetrov) [inline makefile:22]: will this work on windows?

File: makefile [hunk 2]
```diff
@@ -1113,11 +1117,15 @@ endef
 $(foreach t,$(releasetbb.LIBS_Y),$(eval $(call .release.t,$t,$(RELEASEDIR.tbb.soia))))
 $(foreach t,$(releasetbb.LIBS_A),$(eval $(call .release.t,$t,$(RELEASEDIR.tbb.libia))))
 
+#----- cmake configs generation
+_release_cmake_configs:
+	$(if $(shell bash -c "command -v cmake"),cmake -DINSTALL_DIR=$(RELEASEDIR.lib)/cmake/oneDAL -P cmake/scripts/generate_config.cmake,echo 'cmake configs generation skipped')
```
Comment (napetrov) [inline makefile:1122]: cmake generation is part of internal infra currently - corresponding changes should be made there

File: makefile (hunk 1 above)
Comment (Alexsandruss) [reply makefile:22]: yes, msys2/bash is requirement for building on Windows and it returns correct PLAT

File: makefile (hunk 2 above)
Comment (Alexsandruss) [reply makefile:1122]: Is there any problem if configs are generated twice? I think auto generation of configs will be more useful for users

Comment (napetrov) [reply makefile:1122]: Agree that it should be generated here. I'm not sure what would happen on second call - if it works - then fine

Comment (Alexsandruss) [reply makefile:1122]: It works, same configs are generated each time

**Pull request merged**